pandas/core/arrays/integer.py

```diff
@@ -529,6 +529,55 @@ def cmp_method(self, other):
         name = '__{name}__'.format(name=op.__name__)
         return set_function_name(cmp_method, name, cls)
 
+    def _reduce(self, op, name, axis=0, skipna=True, numeric_only=None,
+                filter_type=None, **kwds):
+        """Return a scalar result of performing the op
+
+        Parameters
+        ----------
+        op : callable
+            function to apply to the array
+        name : str
+            name of the function
+        axis : int, default 0
+            axis over which to apply, defined as 0 currently
+        skipna : bool, default True
+            if True, skip NaN values
+        numeric_only : bool, optional
+            if True, only perform numeric ops
+        filter_type : str, optional
+        kwds : dict
+
+        Returns
+        -------
+        scalar
+        """
+
+        data = self._data
+        mask = self._mask
+
+        # coerce to a nan-aware float if needed
+        if mask.any():
```

Comment:
so this doesn't work, because we *only* allow the ``skipna`` parameter to ``nanops``, rather than a mask (e.g. it computes its own mask), which if we don't set to np.nan doesn't work. 

So I think this could be added (e.g. allow skipna as a mask itself) to just use that.

Reply:
see my comment above, this is not possible as all of the special casing for nan removals is already in nanops, and silly to duplicate here.